### PR TITLE
[ADO] Additional telemetry

### PR DESCRIPTION
**Pull request by Garima-msft**
## Description
</br>
Add additional telemetry for:
1. Time taken for each resource evaluation
2. Time taken for each control evaluation
3. Send resource wise count when scan starts
4. Time taken by each listener when scan is completed

Additionally, modifications to methods used for publishing csv and printing summary

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Abstracts/ServicesSecurityStatus.ps1 [hunk 1]
```diff
@@ -189,6 +217,24 @@ class ServicesSecurityStatus: ADOSVTCommandBase
...
+						TimeTakenInMs = $this.StopWatch.ElapsedMilliseconds;
+						ResourceCount = "$currentCount/$totalResources"; 
+						ResourceName = $svtObject.ResourceContext.ResourceName;
+						ResourceType = $svtObject.ResourceContext.ResourceType ;
+						ScanStartDateTime = $this.ScanStart;
+						ScanEndDateTime = $this.ScanEnd;
+						RunIdentifier = $this.RunIdentifier;
+					}
+
+					[AIOrgTelemetryHelper]::PublishEvent( "ADOScanner additional telemetry",$properties, @{})
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Abstracts/ServicesSecurityStatus.ps1:236]: resource scan completed

File: src/AzSK.Framework/Abstracts/ListenerBase.ps1 [hunk 2]
```diff
@@ -54,6 +55,47 @@ class ListenerBase: EventBase
              $this.PublishException($_);
         }
     }
+    
+    [void] PushAIEvents([String] $Eventname)
```
Comment (Garima-msft) [inline src/AzSK.Framework/Abstracts/ListenerBase.ps1:59]: comment 
used only for perf analysis internally in ADO context

Comment (Garima-msft) [reply src/AzSK.Framework/Abstracts/ListenerBase.ps1:59]: method name : PushAIEventsfromHandler

File: src/AzSK.Framework/Abstracts/ListenerBase.ps1 [hunk 3]
```diff
@@ -54,6 +55,47 @@ class ListenerBase: EventBase
...
+                'time' = ([System.dateTime]::UtcNow.ToString('o'))
+                'iKey' = $iKey
+                'tags' = [PSCustomObject]@{
+                    'ai.internal.sdkVersion' = 'dotnet: 2.1.0.26048'
+                }
+                'data' = [PSCustomObject]@{
+                    'baseType' = 'EventData'
+                    'baseData' = [PSCustomObject]@{
+                        'ver' = '2'
+                        'name' = "ADOScanner additional telemetry"
```
Comment (Garima-msft) [inline src/AzSK.Framework/Abstracts/ListenerBase.ps1:85]: Post Command Completion Events

File: src/AzSK.Framework/Abstracts/SVTBase.ps1 [hunk 4]
```diff
@@ -644,10 +646,16 @@ class SVTBase: AzSKRoot
 				#Getting scan time for each control in Staging environment
 				if ($this.invocationContext.MyCommand.ModuleName -match "staging")
```
Comment (Garima-msft) [inline src/AzSK.Framework/Abstracts/SVTBase.ps1:647]: remove staging specific check

File: src/AzSK.Framework/Listeners/FixControl/WriteFixControlFiles.ps1 [hunk 5]
```diff
@@ -47,7 +47,8 @@ Repair-AzSKSubscriptionSecurity `
         });
 
         $this.RegisterEvent([SVTEvent]::CommandCompleted, {
-            $currentInstance = [WriteFixControlFiles]::GetInstance();
+			$currentInstance = [WriteFixControlFiles]::GetInstance();
+            $currentInstance.PushAIEvents("WriteFixControlFiles CommandCompleted");
```
Comment (Garima-msft) [inline src/AzSK.Framework/Listeners/FixControl/WriteFixControlFiles.ps1:51]: Check is this available here: 
[ListenerBase]::isAIKeyEnabled

File: src/AzSK.Framework/Listeners/RemoteReports/AIOrgTelemetry.ps1 [hunk 6]
```diff
@@ -218,7 +218,9 @@ class AIOrgTelemetry: ListenerBase {
 				$properties.Add("HasRequiredAccess", $results[0].CurrentSessionContext.Permissions.HasRequiredAccess)
 				if ($this.invocationContext.MyCommand.ModuleName -match "staging")
```
Comment (Garima-msft) [inline src/AzSK.Framework/Listeners/RemoteReports/AIOrgTelemetry.ps1:219]: remove staging check
change to ai telemetry check

**Review by abhaydaga — APPROVED** [review]

**Pull request merged**